src/lib/ERC4337/interface/IAccount.sol

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import {UserOperation} from "src/lib/ERC4337/utils/UserOperation.sol";
+
+/// @title ERC-4337 IAccount Interface
+/// @author Original EIP-4337 Spec Authors: https://eips.ethereum.org/EIPS/eip-4337
+
+/// @dev Interface contract taken from the original EIP-4337 spec,
+/// used to signify ERC-4337 compliance for smart account wallets inheriting from this contract
+interface IAccount {
+
+    function validateUserOp(
+        UserOperation calldata userOp, 
+        bytes32 userOpHash, 
+        uint256 missingAccountFunds
+    )  external returns (uint256 validationData);
+}
```

Comment:
This interface is the one directly defined by the EIP4337 spec so I'm reluctant to change it. I see two options:

1.  Create a separate `IAccounts` interface that contains the extra functions beyond the 4337 spec
2.  Create an abstract contract that inherits the spec and adds the extra functions

Option 1 has the unfortunate drawback of delinearizing the inheritance tree and has confusing naming (though a better name could be chosen). I like option 2 since it's what most other repos I've seen do, usually calling it something like `BaseAccount`, and is also generally what I was aiming for with the `Account` abstract contract. But now that `Account` is a hub for a number of other functionalities I agree it would be cleaner to put those functions into some intermediary abstract contract or interface that itself inherits the IAccount spec.

Reply:
Resolved using option #2 in 6dba6f5